src/media/media.ts

```diff
@@ -308,22 +313,27 @@ export class SdkMedia extends (EventEmitter as { new(): StrictEventEmitter<Event
     mediaReqOptions: IMediaRequestOptions = { video: true, audio: true },
     retryOnFailure = true
   ): Promise<MediaStream> {
+    const optionsCopy = { ...mediaReqOptions };
+    if (!optionsCopy.uuid) {
+      optionsCopy.uuid = v4();
+    }
+
     /* `getStandardConstraints` will set media type to `truthy` if `null` was passed in */
-    const requestingVideo = mediaReqOptions.video || mediaReqOptions.video === null;
-    const requestingAudio = mediaReqOptions.audio || mediaReqOptions.audio === null;
-    const conversationId = mediaReqOptions.session?.conversationId;
-    const sessionId = mediaReqOptions.session?.id;
+    const requestingVideo = optionsCopy.video || optionsCopy.video === null;
+    const requestingAudio = optionsCopy.audio || optionsCopy.audio === null;
+    const conversationId = optionsCopy.session?.conversationId;
+    const sessionId = optionsCopy.session?.id;
     const {
       micPermissionsRequested,
       cameraPermissionsRequested
     } = this.getState();
 
-    if (typeof mediaReqOptions.retryOnFailure !== 'boolean') {
-      mediaReqOptions.retryOnFailure = retryOnFailure;
+    if (typeof optionsCopy.retryOnFailure !== 'boolean') {
+      optionsCopy.retryOnFailure = retryOnFailure;
     }
 
     const loggingExtras = {
-      mediaReqOptions: { ...mediaReqOptions, session: undefined },
+      mediaReqOptions: { ...optionsCopy, session: !!optionsCopy.session },
```

Comment:
is `session` supposed to be a boolean?

Reply:
It is just for logging purposes. Before I was just wiping it out by setting it to `session: undefined`. But I thought it may be useful in the logs to at least show if there was a session passed into the function.